### PR TITLE
(aws & titus) Deleting tags from server group upon creation

### PR DESCRIPTION
**Pull request by jeyrschabu**
- Added a createServerGroupEventHandler to handle create server group events
- Deletes tags from servergroup name on deploy

### CODE REVIEW

File: clouddriver-elasticsearch/src/test/groovy/com/netflix/spinnaker/clouddriver/elasticsearch/events/CreateServerGroupEventHandlerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import spock.lang.Unroll
+
+class CreateServerGroupEventHandlerSpec extends Specification {
+  def serverGroupTagger = Mock(ElasticSearchServerGroupTagger)
+
+  @Subject
+  def eventHandler = new CreateServerGroupEventHandler(serverGroupTagger)
+
+  @Unroll
+  void "should only handle DeleteServerGroupEvent"() {
```
Comment (anotherchrisberry) [inline clouddriver-elasticsearch/src/test/groovy/com/netflix/spinnaker/clouddriver/elasticsearch/events/CreateServerGroupEventHandlerSpec.groovy]: s/DeleteServerGroupEvent/CreateServerGroupEvent

Comment (jeyrschabu) [inline clouddriver-elasticsearch/src/test/groovy/com/netflix/spinnaker/clouddriver/elasticsearch/events/CreateServerGroupEventHandlerSpec.groovy]: yup good catch, sir

File: clouddriver-elasticsearch/src/test/groovy/com/netflix/spinnaker/clouddriver/elasticsearch/events/CreateServerGroupEventHandlerSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    given:
+    def operationEvent = Mock(OperationEvent) {
+      getType() >> { return OperationEvent.Type.SERVER_GROUP }
+      getAction() >> { return OperationEvent.Action.CREATE }
```
Comment (anotherchrisberry) [inline clouddriver-elasticsearch/src/test/groovy/com/netflix/spinnaker/clouddriver/elasticsearch/events/CreateServerGroupEventHandlerSpec.groovy]: That this case (type is `SERVER_GROUP`, action is `CREATE`) is not handled is a little confusing - I would kind of expect the event to be handled based on those two fields. But I guess it's good that the test shows it really has to be an event of that particular class.

Maybe update the name of the test to add that clarification? Or not, it's probably fine.

**Comment by anotherchrisberry** [comment]
Couple of very minor comments on the test but LGTM.

**Pull request merged**